client/my-sites/pages/page-list.jsx

```diff
@@ -203,37 +203,40 @@ var Pages = React.createClass( {
 
 		// pages have loaded, sites have loaded, and we have a site instance or are viewing all-sites
 		if ( pages.length && this.props.sites.initialized ) {
-			if ( ! this.props.search ) {
-				// we're listing in reverse chrono. use the markers.
-				pages = this._insertTimeMarkers( pages );
-			}
-			rows = pages.map( function( page ) {
-				if ( ! ( 'site_ID' in page ) ) {
-					return page;
+			const site = this.props.sites.getSelectedSite();
+			const status = this.props.status || 'published';
+
+			// @TODO check if it is the first and last page (means there are fewer than 100)
+			if ( site && status === 'published' && ! this.props.search && this.props.lastPage ) {
+				pages = sortPagesHierarchically( pages );
+
+				rows = pages.map( function( page ) {
+					return (
+						<Page key={ 'page-' + page.global_ID } page={ page } site={ site }
+							multisite={ false } hierarchical={ true } hierarchyLevel={ page.indentLevel || 0 } />
+					);
+				}, this );
```

Comment:
Passing of `this` not needed, as context is never read within the anonymous function. Edit: same goes for other loops in the module.

Tip: for any other case where binding of `this` _is_ necessary, ES6 arrow functions take care of that:

```js
collection.map( item => this.process( item ) )
// equivalent to
collection.map( function( item ) { return this.process( item ); }.bind( this ) )
```